src/silx/gui/data/DataViews.py

```diff
@@ -1066,13 +1068,128 @@ def createWidget(self, parent):
         widget.setDefaultColormap(self.defaultColormap())
         widget.getColormapAction().setColormapDialog(self.defaultColorDialog())
         widget.getIntensityHistogramAction().setVisible(True)
+        widget.getAggregationModeAction().setVisible(True)
+        widget.getAggregationModeAction().sigAggregationModeChanged.connect(
+            self._aggregationModeChanged
+        )
         widget.setKeepDataAspectRatio(True)
         widget.getXAxis().setLabel("X")
         widget.getYAxis().setLabel("Y")
         maskToolsWidget = widget.getMaskToolsDockWidget().widget()
         maskToolsWidget.setItemMaskUpdated(True)
         return widget
 
+    def _aggregationModeChanged(self):
+        plot = self.getWidget()
+        item = plot._getItem("image")
+        if item is None:
+            return
+        
+        aggregationMode = plot.getAggregationModeAction().getAggregationMode()
+        if aggregationMode is not None and isinstance(item, ImageDataAggregated):
+            item.setAggregationMode(aggregationMode)
+        else:
+            image = item.getData(copy=False)
+            if image is None:
+                return
+            origin = item.getOrigin()
+            scale = item.getScale()
+            self.setAggregatedImage(image, origin, scale, copy=False, resetzoom=False)
```

Comment:
Since the view controls its content, the image item could always be an `ImageDataAggregated` and so this code could be simplified.